### PR TITLE
feat: New template structure

### PR DESCRIPTION
**Pull request by fussel178**
## Summary

First setup for the new and discussed Telestion application structure.

## Details

Updated things:
- merge Java Apps and PSCs into one repository (see https://github.com/TelestionTeam/telestion-rocketsound/pull/1)
- add test workflow for core app
- improved Docker publish workflow
  - now uses the new GitHub Container Registry to push Docker images users can use without access to the repo
  - automatic generation from commit messages via [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- switch to Java 16
- add some default/nice-to-have configuration files
- updated READMEs

### Intended Release Cycle

When you want to release a new application, go to the _Actions_ Tab in the GitHub UI and choose the `Release` Action.
Then click `Run workflow`.

image

This triggers the Release Action which automatically creates a Conventional Commit Release on GitHub.
Afterwards the subsequent build actions are triggered via:
```yml
on:
  workflow_run:
    workflows: ["Release"]
    types: [completed]
```

Additionally, if you want to upload release assets in these build workflows, you can use the cached build environment:
```yml
      - name: Restore cache release upload URL ♻️
        uses: actions/cache@v2.1.5
        with:
          path: ~/.build-env
          key: 'github-release-action'

      - name: Import environment
        run: cat ~/.build-env >> $GITHUB_ENV

      - name: Upload release asset
        uses: actions/upload-release-asset@v1.0.1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          upload_url: ${{ env.upload_url }}
          asset_path: ./dist/artifact1.txt
          asset_name: artifact1.txt
          asset_content_type: text/plain
```

You can find more information and an example here:
https://github.com/fussel178/conventional-release-dispatch

Feel free to comment and make improvements on this PR.
I'm open for suggestions. :wink:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,18 +2,20 @@
 
 This is a template for Telestion applications.
 It helps you to set up the publishing of Telestion applications.
-Dependencies to the [main repo](https://github.com/TelestionTeam/telestion) are included, too.
+Dependencies to the [main repo](https://github.com/wuespace/telestion) are included, too.
```
Comment (pklaschka) [inline README.md:5]: `wuespace/telestion` won't be the same as `TelestionTeam/telestion`. The equivalent will, most likely, be called `telestion-core` or something like that...

File: gui/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Telestion CLient PSC
```
Comment (pklaschka) [inline gui/README.md:1]:
```suggestion
# Telestion Client PSC
```

File: .github/workflows/publish-core-image.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      # Temporary fix for growing caches
+      # https://github.com/docker/build-push-action/issues/252
+      # https://github.com/moby/buildkit/issues/1896
+      #
+      # Please see:
+      # https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md#github-cache
+      - name: Move cache ♻️
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
```
Comment (pklaschka) [inline .github/workflows/publish-core-image.yml:97]: Concerning https://github.com/wuespace/telestion-client/issues/521: How would/could this (if initialized) integrate with release workflows for the `gui`?

The thing is that this action already "takes away" the workflow run for releases, and by running the conventional changelog action, already creates a release, meaning the PSC, as far as I can see it, has no place to "inject" its release actions. As far as I know, release runs can't be triggered by releases created in another action...

Comment (fussel178) [reply .github/workflows/publish-core-image.yml]: Hopefully fixed with https://github.com/TelestionTeam/template-telestion-application/pull/2/commits/e30dff30c5e560a094baef346879cf56910285e0.
For more information please see the PR details above.

File: gui/telestion-project [hunk 4]
```diff
@@ -0,0 +1,3 @@
+This file is used to detect the folder structure within our automation tools.
```
Comment (jvpichowski) [inline gui/telestion-project:1]: Move this file to the project root

File: core/settings.gradle [hunk 5]
```diff
@@ -0,0 +1,4 @@
+rootProject.name = 'TelestionTemplate'
```
Comment (jvpichowski) [inline core/settings.gradle:1]: Telestion Project Template

File: core/build.gradle [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+application {
+    mainClassName = 'org.telestion.application.Telestion'
+}
+
+
+repositories {
+    mavenCentral()
+    maven {
+        name = "GitHubPackages"
+        url = uri("https://maven.pkg.github.com/TelestionTeam/telestion/")
```
Comment (jvpichowski) [inline core/build.gradle:22]: Please add a TODO to change this to WueSpace

File: core/build.gradle [hunk 7]
```diff
@@ -0,0 +1,51 @@
+plugins {
+    id 'java'
+    id 'maven-publish'
+    id 'application'
+}
+
+group 'org.telestion.telestiontemplate'
```
Comment (jvpichowski) [inline core/build.gradle:7]: `de.wuespace.telestion.project.template`

File: core/build.gradle [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+version '1.0-SNAPSHOT'
+
+sourceCompatibility = gradle.sourceCompatibility
+targetCompatibility = gradle.targetCompatibility
+
+application {
+    mainClassName = 'org.telestion.application.Telestion'
```
Comment (jvpichowski) [inline core/build.gradle:14]: Also change package to de.wuespace.telestion.Main

File: core/Dockerfile [hunk 9]
```diff
@@ -5,7 +5,7 @@ ENV WORK_DIR /usr/telestion
 EXPOSE 8080
 
 COPY conf/config.json $WORK_DIR/config.json
-COPY build/distributions/RocketSound-*.tar $WORK_DIR/
+COPY build/distributions/TelestionTemplate-*.tar $WORK_DIR/
```
Comment (jvpichowski) [inline core/Dockerfile:8]: We have to check how this works with the spaces

**Review by jvpichowski — COMMENTED** [review]
LGTM

**Comment by jvpichowski** [comment]
core -> application
gui -> client

File: core/Dockerfile (hunk 9 above)
Comment (fussel178) [reply core/Dockerfile]: Fixed in https://github.com/wuespace/telestion-project-template/pull/2/commits/b557f42f81d50a1f13cee5bd0a1aab333ee1bc77

File: core/build.gradle (hunk 8 above)
Comment (fussel178) [reply core/build.gradle]: Fixed in https://github.com/wuespace/telestion-project-template/pull/2/commits/b557f42f81d50a1f13cee5bd0a1aab333ee1bc77

File: core/build.gradle (hunk 7 above)
Comment (fussel178) [reply core/build.gradle]: Fixed in https://github.com/wuespace/telestion-project-template/pull/2/commits/b557f42f81d50a1f13cee5bd0a1aab333ee1bc77

File: core/build.gradle (hunk 6 above)
Comment (fussel178) [reply core/build.gradle]: Fixed in https://github.com/wuespace/telestion-project-template/pull/2/commits/b557f42f81d50a1f13cee5bd0a1aab333ee1bc77

File: core/settings.gradle (hunk 5 above)
Comment (fussel178) [reply core/settings.gradle]: Fixed in https://github.com/wuespace/telestion-project-template/pull/2/commits/b557f42f81d50a1f13cee5bd0a1aab333ee1bc77

File: gui/telestion-project (hunk 4 above)
Comment (fussel178) [reply gui/telestion-project]: Fixed in https://github.com/wuespace/telestion-project-template/pull/2/commits/b557f42f81d50a1f13cee5bd0a1aab333ee1bc77

File: gui/README.md (hunk 2 above)
Comment (fussel178) [reply gui/README.md]: Fixed in https://github.com/wuespace/telestion-project-template/pull/2/commits/b557f42f81d50a1f13cee5bd0a1aab333ee1bc77

File: README.md (hunk 1 above)
Comment (fussel178) [reply README.md]: Fixed in https://github.com/wuespace/telestion-project-template/pull/2/commits/b557f42f81d50a1f13cee5bd0a1aab333ee1bc77

**Comment by fussel178** [comment]
@jvpichowski and @pklaschka Thank you very much for your contributions! :tropical_drink:

**Pull request merged**